kolibri/core/assets/src/vue/content-renderer/download-button.vue

```diff
@@ -58,20 +58,16 @@
       downloadMediaText() {
         return this.$tr('downloadMedia');
       },
+      downloadLink() {
+        const filePath = this.availableFiles[0].storage_url;
+        // Extract filename from the storage url.
+        const filename = filePath.replace(/^.*[\\\/]/, '');
+        const newFilename = this.sanitizeFilename(this.title);
+        return `/downloadcontent/${filename}/${newFilename}`;
+      },
     },
     methods: {
       /**
-      * Method that downloads the content.
-      */
-      downloadContent() {
-        const sanitizedFilename = this.sanitizeFilename(this.title);
-        const x = new XMLHttpRequest();
-        x.open('GET', this.availableFiles[0].storage_url, true);
-        x.responseType = 'blob';
-        x.onload = (e) => downloadjs(e.target.response, sanitizedFilename, this.contentType);
-        x.send();
-      },
-      /**
       * Method that returns a safe filename.
       */
       sanitizeFilename(filename) {
```

Comment:
May not be necessary to be so aggressive in the sanitization. A few things, like slashes, quotes, and colons, likely need to be replaced, but we don't need to fully "sluggify".